invenio_drafts_resources/services/records/components/base.py

```diff
@@ -186,6 +128,61 @@ def new_version(self, identity, draft=None, record=None):
         record_files = self.get_record_files(record)
         draft_files.enabled = record_files.enabled
 
+    def _assign_files_enabled(self, identity, data, draft, errors):
+        """Assign files.enabled."""
+        enabled = data.get(self.files_data_key, {}).get(
+            "enabled", self.service.config.default_files_enabled
+        )
+        if self._check_can_manage(identity, data, draft, errors, enabled):
```

Comment:
minor: this if could be used to create a early return with 
```
if not self._check_can_manage(identity, data, draft, errors, enabled):
    return
```

this would improve readability by removing an indentation level